src/libraries/System.IO.Compression.ZipFile/src/System/IO/Compression/ZipFileExtensions.ZipArchiveEntry.Extract.cs

```diff
@@ -83,7 +83,7 @@ public static void ExtractToFile(this ZipArchiveEntry source, string destination
             {
                 File.SetLastWriteTime(destinationFileName, source.LastWriteTime.DateTime);
```

Comment:
We could make this more resilient if we changed the implementation to set the last write time in ExtractExternalAttributes.  That would use our handle that was already opened.  Any reason not to do this?

Reply:
@adamsitnik proposed a better fix like that [here](https://github.com/dotnet/runtime/pull/63912#pullrequestreview-855217606). Once that linked PR is merged, a better fix like that can probably go in for .NET 7. That linked PR is probably too much to consider for servicing, though.

Reply:
I see.  looks like sharing source and pinvoking was already discarded due to cost/risk in servicing.  At the least, let's make sure there is an issue tracking the better fix: https://github.com/dotnet/runtime/pull/63912#issuecomment-1019876856

Reply:
@ericstj good point, I've created an issue and added it to the 7.0 milestone https://github.com/dotnet/runtime/issues/64288